force-app/main/default/classes/ExtensionControllerTest.cls

```diff
@@ -0,0 +1,142 @@
+    private static void testsetPrimarySuccess() {
+        initData('testMethodNameSuccess');
+        
+         Account refAcct = [SELECT Id,Name FROM Account WHERE Name = 'TestAccount'];
+         Contact refCon = [SELECT Id, LastName, Is_Primary_Contact__c FROM Contact WHERE LastName = 'TestCon2'];
+        
+               //call the current page for testing and attaching values
+               PageReference testPage = Page.PrimaryContact; 
+               Test.setCurrentPage(testPage);
+               testPage.getParameters().put('Id', String.valueOf(refAcct.Id));
+               testPage.getParameters().put('contactId', String.valueOf(refCon.Id));
+
+                
+         //create a standard controller and pass the dummy account controller
+        ApexPages.StandardController accountController = new ApexPages.StandardController(refAcct);
+
+        //Get the Standard Controller and pass it to controller extension
+        extentionController ext = new extentionController(accountController);
+        
+        Test.startTest();
+        //Testing
+        PageReference ref = ext.setPrimary();
+        
+        Test.stopTest();
+        
+        //Assertions
+        Contact assertCon = [Select LastName, Is_Primary_Contact__c FROM Contact WHERE LastName='TestCon2'];
+        System.assert(assertCon.Is_Primary_Contact__c == true);
+        System.assertEquals('/apex/PrimaryContact?id='+refAcct.Id, ref.getUrl());
+
+    }
+    
+    //Testing search() method
+    @isTest 
+    private static void testsearchSuccess()
+    {   
+        initData('testMethodNameSuccess');
+        Account searchAcc = [SELECT Id, Name FROM Account Where Name = 'TestAccount'];
+        
+        List<Contact> Cons = [Select Id, LastName, Is_Primary_Contact__c FROM Contact WHERE LastName Like '%TestCon%'];
+        Map<Integer, Contact> ContMap = new Map<Integer, Contact>();
+                for(Integer index = 0 ; index < Cons.size(); index++) {
+                    ContMap.put(index, Cons[index]);}
+         
+        String sText = 'TestC';
+        
+        PageReference searchPage = Page.PrimaryContact; 
+               Test.setCurrentPage(searchPage);
+               searchPage.getParameters().put('Id', String.valueOf(searchAcc.Id));
+               searchPage.getParameters().put('searchText',sText);
+               
+        //create a standard controller and pass the dummy account controller
+        ApexPages.StandardController accountController = new ApexPages.StandardController(searchAcc);
+
+        //Get the Standard Controller and pass it to controller extension
+        extentionController ext = new extentionController(accountController);
+        
+        //setting ContactMap to dummy values
+        ext.ContactMap = ContMap;
+        
+        //Test
+        Test.startTest();
+        
+        PageReference ref = ext.search();
+        
+        Test.stopTest();
+        
+        //Assert
+        System.assertEquals(null, ref);
```

Comment:
please assert some result, not "null". For example you can check if you have "n" number of contacts